src/main/java/com/minecolonies/coremod/entity/ai/citizen/cook/EntityAIWorkCook.java

```diff
@@ -281,8 +359,25 @@ private AIState serveFood()
         return getState();
     }
 
+    /**
+     * Find a furnace and cook raw food in it.
+     *
+     * If no furnaces, report and transition to START_WORKING.
+     * If we have no cookable food and either we are going nowhere || where we are going is not a furnace, COOK_GATHERING some kind of cookable food
+     * if we are going nowhere, find an empty furnace and walk to it, then repeat this state.
+     * If there are no empty furnaces, transition to START_WORKING.
+     * If we are at a furnace, attempt to put our uncooked food in it.
+     * If the furnace has no fuel and we have no fuel, then COOK_RETRIEVE_FUEL
+     * If the furnace has no fuel and we have fuel, then put fuel in and transition to START_WORKING.
+     * If the furnace has fuel, transition to START_WORKING.
+     * If we walked to the furnace and it's not there, then COOK_COOK_FOOD.
+     *
+     * @return next AIState
+     */
     private AIState cookFood()
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Reduce the number of returns of this method 7, down to the maximum allowed 6. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1142)